server/src/main/java/io/crate/types/FloatType.java

```diff
@@ -58,21 +58,32 @@ public String getName() {
     }
 
     @Override
-    public Float value(Object value) {
+    public Float implicitCast(Object value) throws IllegalArgumentException, ClassCastException {
         if (value == null) {
             return null;
-        }
-        if (value instanceof Float) {
```

Comment:
Might still be worth to keep this early as a "common case" - assuming the input is likey to already be a `Float` (not sure if that is still really the case often)